### PR TITLE
JESI-2600

### PR DESCRIPTION
**Pull request by jessebridge**
[JESI-2600](https://jesims.atlassian.net/browse/JESI-2600)

https://github.com/jesims/jesi-web/pull/637

### CODE REVIEW

File: test/io/jesi/backpack/string_test.cljc [hunk 1]
```diff
@@ -120,3 +123,21 @@
...
+               (bp/remove-prefix (rnd/string) fact))))
+
+      (testing "removes the prefix with a default separator of `, `"
+        (is (= "scorpions, small lizards, snakes, eggs."
+               (bp/remove-prefix "Meerkats hunt and eat insects" fact))))
+
+      (testing "allows specifying a custom separator"
+        (is (= "insects, scorpions, small lizards, snakes, eggs."
+               (bp/remove-prefix "Meerkats hunt and eat" \space fact)))))))
+
```
Comment (AndreTheHunter) [inline test/io/jesi/backpack/string_test.cljc]: 👾

File: src/io/jesi/backpack/string.cljc [hunk 2]
```diff
@@ -7,6 +7,19 @@
...
 
+(defn remove-prefix
+  "Removes the prefix if the string starts with it otherwise ignores, is case sensitive"
+  ([prefix s]
+   (remove-prefix prefix ", " s))
+
+  ([prefix separator s]
+   (let [match (str prefix separator)]
+     (if (string/starts-with? s match)
+       (string/replace-first s match "")
```
Comment (AndreTheHunter) [inline src/io/jesi/backpack/string.cljc]: Would be faster to use `subs` (http://clojuredocs.org/clojure.core/subs):
```suggestion
       (subs s (dec (count match)))
```

File: src/io/jesi/backpack/string.cljc [hunk 3]
```diff
@@ -7,6 +7,19 @@
...
+       (if (= separator ", ")
+         (remove-prefix prefix \space s)
```
Comment (AndreTheHunter) [inline src/io/jesi/backpack/string.cljc]: Why is this needed?

File: src/io/jesi/backpack/string.cljc [hunk 4]
```diff
@@ -15,10 +15,8 @@
   ([prefix separator s]
    (let [match (str prefix separator)]
      (if (string/starts-with? s match)
-       (subs s (dec (count match)))
-       (if (= separator ", ")
-         (remove-prefix prefix \space s)
-         s)))))
+       (subs s (dec (+ 1 (count match))))
```
Comment (jessebridge) [inline src/io/jesi/backpack/string.cljc]: Had to add 1 onto the count so it would add spaces at the start of the string

File: src/io/jesi/backpack.cljc [hunk 5]
```diff
@@ -67,7 +67,8 @@
    subs-to
    suffix
    true-string?
-   uuid-str?]
+   uuid-str?
+   remove-prefix]
```
Comment (axrs) [inline src/io/jesi/backpack.cljc]: sort

File: src/io/jesi/backpack/string.cljc (hunk 4 above)
Comment (axrs) [reply src/io/jesi/backpack/string.cljc]: `dec` stands for decrement which means subtract 1
```clojure
(= dec (partial - 1))
